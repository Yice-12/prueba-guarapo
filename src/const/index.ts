export const SPECIE_LIST = [
  "Human",
  "Alien",
  "Humanoid",
  "Poopybutthole",
  "Mythological",
  "Unknown",
  "Animal",
  "Disease",
  "Robot",
  "Cronenberg",
];

export const GENDER_LIST = ["female", "male", "genderless", "unknown"];
export const STATUS_LIST = ["Alive", "Dead", "Unknown"];
