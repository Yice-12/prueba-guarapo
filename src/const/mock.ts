import { configureStore } from "@reduxjs/toolkit";
import { charactersSlice } from "../store/slices/charactersSlice/charactersSlice";

const mockState = {
  characters: {
    characters: [
      {
        id: 1,
        name: "Rick Sanchez",
        status: "Alive",
        species: "Human",
        type: "",
        gender: "Male",
        origin: {
          name: "Earth (C-137)",
          url: "https://rickandmortyapi.com/api/location/1",
        },
        location: {
          name: "Citadel of Ricks",
          url: "https://rickandmortyapi.com/api/location/3",
        },
        image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        episode: ["https://rickandmortyapi.com/api/episode/1"],
        url: "https://rickandmortyapi.com/api/character/1",
        created: "2017-11-04T18:48:46.250Z",
      },
    ],
    loading: false,
    character: {
      id: 1,
      name: "Rick Sanchez",
      status: "Alive",
      species: "Human",
      type: "",
      gender: "Male",
      origin: {
        name: "Earth (C-137)",
        url: "https://rickandmortyapi.com/api/location/1",
      },
      location: {
        name: "Citadel of Ricks",
        url: "https://rickandmortyapi.com/api/location/3",
      },
      image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      episode: ["https://rickandmortyapi.com/api/episode/1"],
      url: "https://rickandmortyapi.com/api/character/1",
      created: "2017-11-04T18:48:46.250Z",
    },
    episodes: [
      {
        id: 1,
        name: "Pilot",
        air_date: "December 2, 2013",
        episode: "S01E01",
        characters: ["https://rickandmortyapi.com/api/character/1"],
        url: "https://rickandmortyapi.com/api/episode/1",
        created: "2017-11-10T12:56:33.798Z",
      },
    ],
    name: "",
  },
};

export const store = configureStore({
  reducer: {
    characters: charactersSlice.reducer,
  },
  preloadedState: mockState,
});
