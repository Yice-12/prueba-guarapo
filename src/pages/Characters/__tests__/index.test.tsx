import { fireEvent, render } from "@testing-library/react";
import { Characters } from "../";
import { Provider } from "react-redux";
import { store } from "../../../const/mock";
import { BrowserRouter } from "react-router-dom";

test("render Characters", () => {
  const { asFragment, container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Characters />
      </BrowserRouter>
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();

  const select = container.querySelector('select[name="gender"]');

  if (select) {
    fireEvent.change(select, { target: { value: "test" } });
  }
});
