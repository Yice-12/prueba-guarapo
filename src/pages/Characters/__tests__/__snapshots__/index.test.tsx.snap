// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Characters 1`] = `
<DocumentFragment>
  <div
    class="characters"
  >
    <div
      class="characters__content"
    >
      <div
        class="characters__content-img"
      >
        <img
          alt="RickAndMorty"
          src="[object Object]"
        />
      </div>
      <div
        class="characters__container-inputs"
      >
        <div
          class="input"
        >
          <div
            class="input__content"
          >
            <img
              alt="loupe"
              src="[object Object]"
            />
            <input
              placeholder="Filter by name..."
              type="text"
            />
          </div>
        </div>
        <div
          class="dropdown"
        >
          <div
            class="dropdown__content"
          >
            <select
              name="species"
            >
              <option
                hidden=""
                value=""
              >
                Species
              </option>
              <option>
                Human
              </option>
              <option>
                Alien
              </option>
              <option>
                Humanoid
              </option>
              <option>
                Poopybutthole
              </option>
              <option>
                Mythological
              </option>
              <option>
                Unknown
              </option>
              <option>
                Animal
              </option>
              <option>
                Disease
              </option>
              <option>
                Robot
              </option>
              <option>
                Cronenberg
              </option>
            </select>
            <div
              class="dropdown__img-arrowDropdown"
            >
              <img
                alt="arrowDropdown"
                src="[object Object]"
              />
            </div>
          </div>
        </div>
        <div
          class="dropdown"
        >
          <div
            class="dropdown__content"
          >
            <select
              name="gender"
            >
              <option
                hidden=""
                value=""
              >
                Gender
              </option>
              <option>
                female
              </option>
              <option>
                male
              </option>
              <option>
                genderless
              </option>
              <option>
                unknown
              </option>
            </select>
            <div
              class="dropdown__img-arrowDropdown"
            >
              <img
                alt="arrowDropdown"
                src="[object Object]"
              />
            </div>
          </div>
        </div>
        <div
          class="dropdown"
        >
          <div
            class="dropdown__content"
          >
            <select
              name="status"
            >
              <option
                hidden=""
                value=""
              >
                Status
              </option>
              <option>
                Alive
              </option>
              <option>
                Dead
              </option>
              <option>
                Unknown
              </option>
            </select>
            <div
              class="dropdown__img-arrowDropdown"
            >
              <img
                alt="arrowDropdown"
                src="[object Object]"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="cardList"
      >
        <div
          class="cardList__content"
        >
          <div
            class="card"
          >
            <div
              class="card__content"
            >
              <img
                src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
              />
              <div
                class="card__description"
              >
                <h1>
                  Rick Sanchez
                </h1>
                <p>
                  Human
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="characters__container-button"
      >
        <div
          class="button"
        >
          <div
            class="button__content"
          >
            <button>
              LOAD MORE
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
